airbyte-integrations/connectors/source-mongodb-internal-poc/src/main/java/io/airbyte/integrations/source/mongodb/internal/cdc/MongoDbCdcInitializer.java

```diff
@@ -0,0 +1,159 @@
+  private final MongoDbDebeziumFieldsUtil mongoDbDebeziumFieldsUtil;
+
+  @VisibleForTesting
+  MongoDbCdcInitializer(final MongoDbDebeziumStateUtil mongoDbDebeziumStateUtil, final MongoDbDebeziumFieldsUtil mongoDbDebeziumFieldsUtil) {
+    this.mongoDbDebeziumStateUtil = mongoDbDebeziumStateUtil;
+    this.mongoDbDebeziumFieldsUtil = mongoDbDebeziumFieldsUtil;
+  }
+
+  public MongoDbCdcInitializer() {
+    this(new MongoDbDebeziumStateUtil(), new MongoDbDebeziumFieldsUtil());
+  }
+
+  /**
+   * Generates the list of stream iterators based on the configured catalog and stream state. This
+   * list will include any initial snapshot iterators, followed by incremental iterators, where
+   * applicable.
+   *
+   * @param mongoClient The {@link MongoClient} used to interact with the target MongoDB server.
+   * @param catalog The configured Airbyte catalog of streams for the source.
+   * @param stateManager The {@link MongoDbStateManager} that provides state information used for
+   *        iterator selection.
+   * @param emittedAt The timestamp of the sync.
+   * @param config The configuration of the source.
+   * @return The list of stream iterators with initial snapshot iterators before any incremental
+   *         iterators.
+   */
+  public List<AutoCloseableIterator<AirbyteMessage>> createCdcIterators(
+                                                                        final MongoClient mongoClient,
+                                                                        final ConfiguredAirbyteCatalog catalog,
+                                                                        final MongoDbStateManager stateManager,
+                                                                        final Instant emittedAt,
+                                                                        final JsonNode config) {
+
+    final Duration firstRecordWaitTime = FirstRecordWaitTimeUtil.getFirstRecordWaitTime(config);
+    final OptionalInt queueSize = MongoUtil.getDebeziumEventQueueSize(config);
+    final String databaseName = config.get(DATABASE_CONFIGURATION_KEY).asText();
+    // WARNING!!! debezium's mongodb connector doesn't let you specify a list of fields to
+    // include, so we can't filter fields solely using the configured catalog. Instead,
+    // debezium only lets you specify a list of fields to exclude. If the fields to exclude
+    // we specify are not equal to all the fields in the source that are not in the
+    // configured catalog then we will be outputting incorrect data.
+    final Set<ExcludedField> fieldsNotIncludedInCatalog = mongoDbDebeziumFieldsUtil.getFieldsNotIncludedInCatalog(catalog, databaseName, mongoClient);
+    final Properties defaultDebeziumProperties = MongoDbCdcProperties.getDebeziumProperties(fieldsNotIncludedInCatalog);
+    final String replicaSet = config.get(REPLICA_SET_CONFIGURATION_KEY).asText();
+    final JsonNode initialDebeziumState =
```

Comment:
How is the `initialDebeziumState` being passed onto the `MongoDbStateManager`? I think there is a bug, during the first sync the  sharedState is going to be null in the state message emitted by `MongoDbStateManager`.